File: frontend/src/types/coin.ts
// src/types/coin.ts
export interface Coin {
  id: string;
  symbol: string;
  name: string;
  image: string;
  current_price: number;
  market_cap: number;
  market_cap_rank: number;
  price_change_percentage_24h: number;
  price_change_percentage_7d_in_currency: number; 
  price_change_percentage_1y_in_currency: number;  
}

export interface PriceHistoryPoint {
  date: string;
  price: number;
}

File: backend/models/coin.go
// backend/models/coin.go
package models

type Coin struct {
	ID            string  `json:"id" gorm:"primaryKey"`
	Symbol        string  `json:"symbol"`
	Name          string  `json:"name"`
	Image         string  `json:"image"`
	CurrentPrice  float64 `json:"current_price"`
	MarketCap     float64 `json:"market_cap"`
	MarketCapRank int     `json:"market_cap_rank"`
	Change24h     float64 `json:"price_change_percentage_24h" gorm:"column:change_24h"`
	Change7d      float64 `json:"price_change_percentage_7d_in_currency" gorm:"column:change_7d"`
	Change1y      float64 `json:"price_change_percentage_1y_in_currency" gorm:"column:change_1y"`
}
